samcli/lib/sync/watch_manager.py

```diff
@@ -227,10 +233,34 @@ def _execute_infra_sync(self) -> None:
             self._observer.unschedule_all()
             self._add_template_triggers()
         else:
+            # Update stacks and repopulate triggers
             # Trigger are not removed until infra sync is finished as there
             # can be code changes during infra sync.
             self._start_sync()
 
+            if not infra_sync_result.infra_sync_executed:
+                # This is for initiating code sync for all resources
+                # To improve: only initiate code syncs for ones with template changes
+                self._queue_up_code_syncs(infra_sync_result.code_sync_resources)
+                LOG.info(self._color.green("Skipped infra sync and queued up required code syncs."))
+            else:
+                LOG.info(self._color.green("Infra sync completed."))
+
+    def _queue_up_code_syncs(self, resource_ids_with_code_sync: Set[ResourceIdentifier]) -> None:
+        """
+        For ther given resource IDs, create sync flow tasks in the queue
+
+        Parameters
+        ----------
+        resource_ids_with_code_sync: Set[ResourceIdentifier]
+            The set of resource IDs to be synced
+        """
+        for resource_id in resource_ids_with_code_sync:
+            if self._sync_flow_factory:
+                sync_flow = self._sync_flow_factory.create_sync_flow(resource_id)
+                if sync_flow:
+                    self._sync_flow_executor.add_delayed_sync_flow(sync_flow)
```

Comment:
nit: Let's use early exit if statements to improve readability?

Reply:
This logic here is consistent with on_code_change() on line 278, consistency improves readability

Reply:
I agree consistency improves readability, but it is not same at line 278. There is no `for` loop there (which is only required here), there is no `if self._sync_flow_factory`. These extra indentations are making this one harder to read.

Reply:
I agree we can early exit on if not self._sync_flow_factory, but for the second if since we have a for loop it doesn't really make a difference since we will need to continue on the for loop anyways